Rework sourcemap troubleshooting session for debug IDs (#7049)
getsentry/sentry-docs: src/platform-includes/sourcemaps/troubleshooting/javascript.mdx

Source maps can sometimes be tricky to get going. If you're having trouble, you can try using our verification tool inside `sentry-cli`, or go through the steps listed below yourself.

> To use the automated verification process, [install and configure](/product/cli/) `sentry-cli`. Once completed, use the `sourcemaps explain` command, by calling it with the event ID that can be found in the top-left corner of the **Issue Details** page in [sentry.io](https://sentry.io). For example, "Event ID: fb71278d".

Reviewer: l: Should we include a screenshot where to find the ID?

Author: Definitely!

```

This should provide you with some useful hints about what exactly went wrong with the source maps setup process.
If it doesn't, let us know so we can make it better.

## Verify artifacts are uploaded

You can find the artifacts uploaded to Sentry by navigating to **[Settings] » Projects » Select your project » Source Maps**.

Make sure all of the necessary files are available. For Sentry to de-minify your stack traces you must provide both the minified files (for example, app.min.js) and the corresponding source maps. In case the source map files do not contain your original source code (`sourcesContent`), you must additionally provide the original source files. Alternatively, sentry-cli will automatically embed the sources (if missing) into your source maps.

## Verify you are building sourcemaps

Bundlers and tools (like `tsc`) that generate code generally require you to manually set options in order for source maps to be generated.

If you followed our tool-specific guides, verify that you configured your tool to emit source maps and that the source maps contain your original source code inside the `sourcesContent` field.

> ## Verify your source files contain debug ID injection snippets

Reviewer: I think the wording makes sense (+ see my comment regarding the code snippet). I'm just wondering for everyone who uses source maps w/o debug ids, this section will be confusing (they'll most likely read about this for the first time and identify that this code snippet isn't in their bundle, causing them to question their own sanity etc... 😅 ).

Should we add a sentence that redirects them to "Uploading without Debug IDs"? 

Something along the lines of 
```suggestion
## Verify your source files contain debug ID injection snippets

If you're uploading source maps by associating them to Sentry commits (e.g. Webpack Plugin v1), see [Uploading Source Maps without Debug IDs](LINK)

```

Author: I purposefully left out any references to the old process in my changes, simply because those who use the old process, likely already have working source maps and don't need the troubleshooting guide. I am honestly not sure if that is actually a good decision. In my mind it was worth a shot, just to reduce confusion/docs-spaghetty overall. Does that make sense?

> Inside the JavaScript files you uploaded to Sentry, search for code that roughly looks like `e._sentryDebugIds=e._sentryDebugIds||{}`. Having this code inside a bundle means that it will be able to be matched to a source file. Every bundle you deploy in your app needs to have this snippet in order to be correctly source mapped.

Reviewer: Any chance that `e` could be minified/mangled to something else in user code? Would it be enough to just let them search for `_sentryDebugIds`?

Author: Had a similar thought while writing this. The reason we can't let them just search for `_sentryDebugIds` is that it's also inside SDK code. Since injection usually happens (or rather _should happen_) very late in the build process, it's likely not minified/mangled further anyhow so I would leave this as is. Wdyt?

<Note>

Chances are, you set up source maps for Sentry a while ago. Since then we updated a lot of our tooling to make the process around source maps easier.
We recommend you upgrade your tooling to the newest versions, as that is usually easier than to troubleshoot the legacy method.

If you still want to find information on the legacy method of uploading source maps, please see the <PlatformLink to="/sourcemaps/troubleshooting_js/uploading-without-debug-ids">guide to upload source maps without debug IDs</PlatformLink>.

</Note>

Source maps can sometimes be tricky to get going. If you're having trouble, you can try using our verification tool inside `sentry-cli`, or go through the steps listed below yourself.

> To use the automated verification process, [install and configure](/product/cli/) `sentry-cli`. Once completed, use the `sourcemaps explain` command, by calling it with the event ID that can be found in the top-left corner of the **Issue Details** page in [sentry.io](https://sentry.io). For example, "Event ID: c2ad049f".

Reviewer 2: FYI, where possible we try to make the linked text be the doc's page title, following Google's style guide: https://developers.google.com/style/link-text

![Image highlighting where to find the ID of an event on Sentry](event_id.png)

```shell
sentry-cli sourcemaps explain c2ad049fd9e448ada7849df94575e019
```

This should provide you with some useful hints about what exactly went wrong with the source maps setup process.
If it doesn't, let us know so we can make it better.

## Verify artifacts are uploaded

> You can find the artifacts uploaded to Sentry by navigating to **[Settings] » Projects » Select your project » Source Maps**.

Reviewer 2: ```suggestion
Verify your artifacts are being uploaded to Sentry. You can find them at **[Settings] > Projects > Select your project > Source Maps**.
```

For Sentry to de-minify your stack traces you must provide both the minified files (for example, app.min.js) and the corresponding source maps.

## Verify you are building sourcemaps

> Bundlers and tools (like `tsc`) that generate code, generally require you to manually set specific options in order for source maps to be generated.

Reviewer 2: ```suggestion
Bundlers and tools (like `tsc`) that generate code, often require you to manually set specific options to generate source maps.
```

> If you followed our tool-specific guides, verify that you configured your tool to emit source maps and that the source maps contain your original source code inside the `sourcesContent` field.

Reviewer 2: ```suggestion
If you followed one of our tool-specific guides, verify you configured your tool to emit source maps and that the source maps contain your original source code in the `sourcesContent` field.
```

## Verify your source files contain debug ID injection snippets

> Inside the JavaScript files you uploaded to Sentry, search for code that roughly looks like `e._sentryDebugIds=e._sentryDebugIds||{}` (note that this snipped might be looking different depending on how you process your code). Having this code inside a bundle means that it will be able to be matched to a source file. Every bundle you deploy in your app needs to have this snippet in order to be correctly source mapped.

Reviewer 2: ```suggestion
In the JavaScript files you uploaded to Sentry, search for code that roughly looks like `e._sentryDebugIds=e._sentryDebugIds||{}`. This code snippet might look different depending on how you process your code.

If this code exists in a bundle, that bundle will be able to be matched to a source file. Every bundle you deploy in your app needs to have this snippet in order to be correctly source mapped.
```

> If your source code does not have the snippet and if you're using a Sentry plugin for your bundler, please check that you are using the latest version and please verify that the plugin is correctly processing your files. You can set the `debug` option to `true` in order to print useful information.

Reviewer 2: ```suggestion
If your source code does not contain this snippet and you're using a Sentry plugin for your bundler, please check that you are using the latest version and please verify that the plugin is correctly processing your files. Set the `debug` option to `true` to print useful degbugging information.
```

> If you are using Sentry CLI, please verify that you are running the `inject` command **before** you're uploading to Sentry and **before** you're deploying your files.

Reviewer 2: ```suggestion
If you're using the Sentry CLI, verify that you're running the `inject` command **before** you upload to Sentry and **before** you deploy your files.
```

## Verify your events contain debug metadata

> When inspecting the event payload JSON data in Sentry, verify that the events have a `debug_meta` property. If this property is missing and you already made sure your source files contain debug ID injection snippets, this likely means that you're using an outdated SDK. We generally recommend upgrading to the latest version, but for source maps to work you need to at least use version `7.47.0`.

Reviewer 2: ```suggestion
When inspecting the event payload JSON data in Sentry, verify that the events have a `debug_meta` property. If this property is missing and you already made sure your source files contain debug ID injection snippets, this likely means that you're using an outdated SDK. 

We generally recommend upgrading to the latest version, and for source maps to work you need to at least use version `7.47.0`.
```

> When you verified that your event has a `debug_meta` property, the next thing to check is whether all frames inside the `raw_stacktrace` property inside the events have an `abs_path` property that exactly matches one of the `code_file` properties inside the `debug_meta` images.

Reviewer 2: ```suggestion
After you verified your event has a `debug_meta` property, the next thing to check is whether all frames inside the `raw_stacktrace` property inside the events have an `abs_path` property that exactly matches one of the `code_file` properties inside the `debug_meta` images.
```
